### PR TITLE
various i386 and amd64 decoding fixes

### PR DESCRIPTION
**Pull request by rakuy0**
It doesn't fix all of the disassembling issues I've found, especially ones like #145 and #153, but it does fix some of the other issues with like psllw and others in i386.

And of course, flake8 fixes abound, and now featuring more unit tests.

### CODE REVIEW

File: envi/archs/amd64/disasm.py [hunk 1]
```diff
@@ -526,13 +523,13 @@ def ameth_c(self, bytes, offset, tsize, prefixes, operflags):
         osize, oper = e_i386.i386Disasm.ameth_c(self, bytes, offset, tsize, prefixes, operflags)
         if prefixes & PREFIX_REX_R:
             oper.reg += REX_BUMP
-        return osize,oper
+        return osize, oper
```
Comment (atlas0fd00m) [inline envi/archs/amd64/disasm.py]: could you simply convert the whole Vivisect over to Flake8 in one PR?  reviewing changes when there are so many white-space diffs is frustrating.  thanks!

File: envi/archs/amd64/opcode64.py [hunk 2]
```diff
@@ -1170,50 +1170,50 @@
...
+tbl32_0F73 = [(0, 0, 0, 0, 0, 0, 0, 0, 0, 0) for x in range(8)]
+tbl32_0F73[2] = (0, INS_OTHER, ADDRMETH_N | OPTYPE_q | OP_W, ADDRMETH_I | OPTYPE_b | OP_R, ARG_NONE, cpu_PENTMMX, "psrlq", 0, 0, 0)
+tbl32_0F73[6] = (0, INS_OTHER, ADDRMETH_N | OPTYPE_q | OP_W, ADDRMETH_I | OPTYPE_b | OP_R, ARG_NONE, cpu_PENTMMX, "psllq", 0, 0, 0)
 
-tbl32_660F73 =  [ ( 0, 0, 0, 0, 0, 0, 0, 0, 0, 0) for x in range(8) ] 
-tbl32_660F73[2] =   ( 0, INS_OTHER, ADDRMETH_V | OPTYPE_dq | OP_W, ADDRMETH_I | OPTYPE_b | OP_R, ARG_NONE, cpu_PENTMMX, "psrlq", 0, 0, 0)
-tbl32_660F73[6] =   ( 0, INS_OTHER, ADDRMETH_V | OPTYPE_dq | OP_W, ADDRMETH_I | OPTYPE_b | OP_R, ARG_NONE, cpu_PENTMMX, "psllq", 0, 0, 0)
-tbl32_660F73[7] =   ( 0, INS_OTHER, ADDRMETH_V | OPTYPE_dq | OP_W, ADDRMETH_I | OPTYPE_b | OP_R, ARG_NONE, cpu_PENTMMX, "psldq", 0, 0, 0)
+tbl32_660F73 = [(0, 0, 0, 0, 0, 0, 0, 0, 0, 0) for x in range(8)]
+tbl32_660F73[2] = (0, INS_OTHER, ADDRMETH_Z | OPTYPE_dq | OP_W, ADDRMETH_I | OPTYPE_b | OP_R, ARG_NONE, cpu_PENTMMX, "psrlq", 0, 0, 0)
```
[the remark this answers is not in the document]
Comment (rakuy0) [reply envi/archs/amd64/opcode64.py]: To support the VEX instructions? Those resolve down into table `tbl32_660f`? These are just for the normal psrlq instructions like psrlq xmm1, 8.

Still a good point though, because something like `vpsrlq xmm1, xmm2, 7` resolves down to tbl32_660f, and it still decodes incorrectly for us, because something like `psrlq  xmm3,xmm4` also resolve down to that same table, and the two instructions have different  handling mnemonics. 

TL;DR: we also need to discuss prefix handling for amd64 disassembling, not just i386.

File: envi/archs/i386/opcode86.py [hunk 3]
```diff
@@ -1091,262 +1091,262 @@
...
+    ( 0, INS_ARITH, ADDRMETH_P | OPTYPE_q | OP_W, ADDRMETH_Q | OPTYPE_q | OP_R, ARG_NONE, cpu_PENTIUM2, "pmaxub", 0, 0, 0),
+    ( 0, INS_AND, ADDRMETH_P | OPTYPE_q | OP_W, ADDRMETH_Q | OPTYPE_q | OP_R, ARG_NONE, cpu_PENTMMX, "pandn", 0, 0, 0),
+        ( 0, INS_OTHER, ADDRMETH_P | OPTYPE_q | OP_W, ADDRMETH_Q | OPTYPE_q | OP_R, ARG_NONE, cpu_PENTIUM2, "pavgb", 0, 0, 0),
+    ( 0, INS_OTHER, ADDRMETH_P | OPTYPE_q | OP_W, ADDRMETH_Q | OPTYPE_q | OP_R, ARG_NONE, cpu_PENTMMX, "psraw", 0, 0, 0),
+    ( 0, INS_OTHER, ADDRMETH_P | OPTYPE_q | OP_W, ADDRMETH_Q | OPTYPE_q | OP_R, ARG_NONE, cpu_PENTMMX, "psrad", 0, 0, 0),
+    ( 0, INS_OTHER, ADDRMETH_P | OPTYPE_q | OP_W, ADDRMETH_Q | OPTYPE_q | OP_R, ARG_NONE, cpu_PENTIUM2, "pavgw", 0, 0, 0),
+    ( 0, INS_MUL, ADDRMETH_P | OPTYPE_q | OP_W, ADDRMETH_Q | OPTYPE_q | OP_R, ARG_NONE, cpu_PENTIUM2, "pmulhuw", 0, 0, 0),
+    ( 0, INS_MUL, ADDRMETH_P | OPTYPE_q | OP_W, ADDRMETH_Q | OPTYPE_q | OP_R, ARG_NONE, cpu_PENTMMX, "pmulhw", 0, 0, 0),
+    (0, INS_OTHER, ADDRMETH_V | OPTYPE_x | OP_W, ADDRMETH_W | OPTYPE_pd | OP_R, ARG_NONE, ARG_NONE, cpu_PENTMMX, "cvtdq2pd", 0, 0, 0),  # Horked due to how we do prefixes
```
[the remark this answers is not in the document]
Comment (rakuy0) [reply envi/archs/i386/opcode86.py]: Agreed.

File: envi/archs/i386/opcode86.py [hunk 4]
```diff
@@ -1091,262 +1091,262 @@
...
+    ( 0, INS_CMP, ADDRMETH_P | OPTYPE_q | OP_W, ADDRMETH_Q | OPTYPE_q | OP_R, ARG_NONE, cpu_PENTMMX, "pcmpeqw", 0, 0, 0),
+    ( 0, INS_CMP, ADDRMETH_P | OPTYPE_q | OP_W, ADDRMETH_Q | OPTYPE_q | OP_R, ARG_NONE, cpu_PENTMMX, "pcmpeqd", 0, 0, 0),
+    (0, INS_OTHER, ARG_NONE, ARG_NONE, ARG_NONE, cpu_PENTMMX, "emms", 0, 0, 0),
+    (0, INS_SYSTEM, ADDRMETH_E | OPTYPE_d | OP_W, ADDRMETH_G | OPTYPE_d | OP_R, ARG_NONE, cpu_PENTIUM2, "vmread", 0, 0, 0),
+    (0, INS_SYSTEM, ADDRMETH_G | OPTYPE_d | OP_W, ADDRMETH_E | OPTYPE_d | OP_R, ARG_NONE, cpu_PENTIUM2, "vmwrite", 0, 0, 0),
+    (0, 0, ARG_NONE, ARG_NONE, ARG_NONE, 0, 0, 0, 0, 0),
+    (0, 0, ARG_NONE, ARG_NONE, ARG_NONE, 0, 0, 0, 0, 0),
+    (0, 0, ARG_NONE, ARG_NONE, ARG_NONE, 0, 0, 0, 0, 0),
+    (0, 0, ARG_NONE, ARG_NONE, ARG_NONE, 0, 0, 0, 0, 0),
+    ( 0, INS_MOV, ADDRMETH_V | OPTYPE_q | OP_W, ADDRMETH_W | OPTYPE_q | OP_R, ARG_NONE, cpu_PENTMMX, "movq", 0, 0, 0),
```
[the remark this answers is not in the document]
Comment (rakuy0) [reply envi/archs/i386/opcode86.py]: Thanks!

File: envi/archs/i386/opcode86.py [hunk 5]
```diff
@@ -525,7 +525,7 @@
 (0, INS_ADD, ADDRMETH_P | OPTYPE_q | OP_W, ADDRMETH_Q | OPTYPE_q | OP_R, ARG_NONE, cpu_PENTMMX, "paddusw", 0, 0, 0),  
 (0, INS_ARITH, ADDRMETH_P | OPTYPE_q | OP_W, ADDRMETH_Q | OPTYPE_q | OP_R, ARG_NONE, cpu_PENTIUM2, "pmaxub", 0, 0, 0),  
 (0, INS_AND, ADDRMETH_P | OPTYPE_q | OP_W, ADDRMETH_Q | OPTYPE_q | OP_R, ARG_NONE, cpu_PENTMMX, "pandn", 0, 0, 0),  
-# 0ff0
+# 0fe0
```
[the remark this answers is not in the document]
Comment (rakuy0) [reply envi/archs/i386/opcode86.py]: :+1:

File: envi/archs/amd64/disasm.py (hunk 1 above)
Comment (rakuy0) [reply envi/archs/amd64/disasm.py]: No automated filter. Just me seeing something I don't like and fixing it along the way.

But yea. I'll convert as much as I can over in another PR. I tried to limit here, but that only went so well XD

File: envi/archs/amd64/opcode64.py [hunk 6]
```diff
@@ -991,7 +991,7 @@
 ( 0, INS_SYSTEM, ADDRMETH_E | OPTYPE_w | OP_W, ARG_NONE, ARG_NONE, cpu_80386, "lmsw", 0, 0, 0),  
 ( 0, INS_SYSTEM, ADDRMETH_E | OPTYPE_w | OP_W, ARG_NONE, ARG_NONE, cpu_80386, "lmsw", 0, 0, 0),  
 ( 0, INS_OTHER, 0, 0, 0, cpu_PENTIUM2, "swapgs", 0, 0, 0),
-( 0, 0, 0, 0, 0, 0, 0, 0, 0, 0),
+(0, INS_SYSTEM, ARG_NONE, ARG_NONE, ARG_NONE, cpu_PENTIUM, "rdtscp", 0, 0, 0),
```
[the remark this answers is not in the document]
Comment (rakuy0) [reply envi/archs/amd64/opcode64.py]: Cool. I'll add those in too.

[the remark this answers is not in the document]
Comment (rakuy0) [reply envi/archs/amd64/opcode64.py]: I have a secondary branch out that's branched from this one to fix the prefix stuff.
I'll put those in there and throw that branch up after this one gets merged.
